### ISSUE TITLE
Unable to reproduce results on Cifar100 with ViT model

### ISSUE THREAD

**Issue by ghost** [jwf40/Information-Theoretic-Unlearning#1]
Hi, thanks for your fantastic work!

I pre-trained a ViT model on Cifar-100 as described in the `Setup` reaching an accuracy of 89% on the test set.
I then tried to run using the checkpoint above the Lipschitz forgetting procedure on the full class:
`python3 src/forget_full_class_main.py -net ViT -dataset Cifar100 -classes 100 -gpu -method lipschitz_forgetting -forget_class 'rocket' -weight_path 'ViT-Cifar100-1-best.pth' -seed 1`

and got the following results: `'TestAcc': 0.9790347814559937, 'RetainTestAcc': 0.9915865659713745, 'df': 0.0. `
Before the forgetting the results were these: `'TestAcc': 89.39873504, 'RetainTestAcc': 89.3966751, 'df': 93.0. `

The Lipschitz loss was the following during weights update:
```
tensor(4.4269, device='cuda:0', grad_fn=<DivBackward0>)
tensor(1.3925, device='cuda:0', grad_fn=<DivBackward0>)
tensor(3.2717, device='cuda:0', grad_fn=<DivBackward0>)
tensor(0.7429, device='cuda:0', grad_fn=<DivBackward0>)
```

I also tried to run with VGG16 for rocket class running following script::
`python3 src/forget_full_class_main.py -net VGG16 -dataset Cifar100 -classes 100 -gpu -method lipschitz_forgetting -forget_class 'rocket' -weight_path 'VGG-Cifar100.pth' -seed 1`
and also getting different results from the paper. 
Before forgetting:`'TestAcc': 65.1206, 'RetainTestAcc': 64.945732 'df': 76.0. `
After forgetting: `'TestAcc': 60.28, 'RetainTestAcc': 60.71, 'df': 0.0. `

Any idea what is going on? 
Thanks!

**Comment by jwf40** [comment]
Hi, 
As discussed in DMs, this seems to be an issue regarding training length. Since neural networks learn smooth functions first, and our loss is inducing smoothness, models stopped too early during training experience performance reduction. I believe we trained ViT for 200 epochs. 

Resolving issue, and I will add links in the README.md to our original weights if you would like to use them!

**Comment by NilakshanKunananthaseelan** [comment]
I also got similar results with VGG16 ('df': 0.0.). The algorithm looks sensitive to the number of iterations used for Lipchitz smoothening.
